src/updatePr.ts

```diff
@@ -21,89 +23,122 @@ declare function schedule<T>(asyncFunction: Promise<T>): void;
 const cleanChangelogRegex =
   /^(fix(\(.+\))?!?: |feat(\(.+\))?!?: |chore(\(.+\))?!?: )?(.*)$/;
 
-export const updatePrTitleAndLabel = (
-  ticketList: ReadonlyArray<GenericTicket>
-): Rr.Reader<RecordScope, void> =>
-  pipe(
-    getTicketsScope(ticketList),
-    Rr.map((e_scope) => {
-      const scope = pipe(
-        e_scope,
-        E.bimap(
-          (err) => warn(err.message),
-          (s) => `(${s})`
-        ),
-        E.fold(
-          () => "",
-          (s) => s
-        )
-      );
+export const updatePrTitleAndLabel =
+  (ticketList: ReadonlyArray<GenericTicket>): Rr.Reader<Configuration, void> =>
+  (configuration) => {
+    const scope = pipe(
+      getTicketsScope(ticketList)(configuration),
+      E.bimap(
+        (err) => warn(err.message),
+        (s) => `(${s})`
+      ),
+      E.fold(
+        () => "",
+        (s) => s
+      )
+    );
 
-      const updateLabel = danger.github.utils.createOrAddLabel({
-        name: scope.replace("(", "").replace(")", ""),
-        // The color is not used and can be customized from the "label" tab in the github page
-        color: "#FFFFFF",
-        description: scope,
-      });
+    const updateLabelAction = pipe(
+      configuration.updateLabel,
+      match(
+        () => TE.left(new Error("updateLabel is disabled")),
+        () =>
+          pipe(
+            TE.tryCatch(
+              async () =>
+                danger.github.utils.createOrAddLabel({
+                  name: scope.replace("(", "").replace(")", ""),
+                  // The color is not used and can be customized from the "label" tab in the github page
+                  color: "#FFFFFF",
+                  description: scope,
+                }),
+              () => {
+                const errorMessage = "Eror during github label update";
```

Comment:
```suggestion
                const errorMessage = "Error during github label update";
```

I don't know where's the entry point, but usually the application code just returns error then, only at top level, you plug them with IO (calling `warn` for example)

Reply:
yes the problem is that the entry point is inside a danger `schedule` function that takes a `promise` as input and executes it. How can I avoid this round? It should be noted that I could also remove the warn since it is not needed

Reply:
where exactly?

Reply:
I propose this solution https://github.com/pagopa/danger-plugin/pull/5/commits/3c0f093a803073174658911bc35c55046a352d46

Reply:
You have to refactor `customRules` method and put all side effects there (`warn`, `markdown`)